.github/workflows/run-build.yaml

```diff
@@ -0,0 +1,254 @@
+          forwards to the runtime do not need to be listed here.
+          See the list of environment variables that are auto-forwarded in
+          the Nextstrain CLI repo (https://github.com/nextstrain/cli/blob/master/nextstrain/cli/hostenv.py)
+
+          The environment variables must be passed via the `env` input or
+          via GitHub Action's secrets that are inherited by using `secrets: inherit`.
+          The environment variable names should be separated by a space,
+          e.g. "SLACK_TOKEN SLACK_CHANNELS".
+        type: string
+        required: false
+
+      cpus:
+        description: >-
+          Number of CPUs/cores/threads/jobs to utilize at once.
+          Will be passed to the `--cpus` option for `nextstrain build` and the
+          `--cores` option for Snakemake. See `nextstrain build` docs for more details:
+          https://docs.nextstrain.org/projects/cli/page/commands/build/
+        type: number
+        default: 2
+        required: false
+
+      memory:
+        description: >-
+          Amount of memory to make available to the build. Passed to the
+          `--memory` option for `nextstrain build`.
+          See `nextstrain build` docs for more details:
+          https://docs.nextstrain.org/projects/cli/page/commands/build/
+        type: string
+        required: false
+
+      build-args:
+        description: >-
+          Additional command-line arguments to pass to `nextstrain build` after
+          the build directory (i.e. additional args for Snakemake).
+        type: string
+        default: ""
+        required: false
+
+      artifact-name:
+        description: >-
+          Name to use for the build output artifact uploaded at end of the workflow.
+
+          If you're invoking this workflow multiple times from the same calling
+          workflow, you should set this. Otherwise, the default "build-outputs"
+          is probably fine.
+        type: string
+        default: build-outputs
+        required: false
+
+      build-output-paths:
+        description: >-
+          List of build output paths to include in the build output artifact uploaded
+          at the end of the workflow, as a string following the format of the
+          `paths` input of the `actions/upload-artifact` action.
+          For example:
+
+            with:
+              build-output-paths: |
+                results/
+                auspice/
+                logs/
+
+          The output will always include a "build.log" file which contains log
+          messages from the `nextstrain build` command. The envdir created to
+          pass environment variables to the build runtime will always be
+          excluded from the output to prevent secrets from being leaked.
+
+          This is not supported for builds on AWS Batch because the workflow
+          detaches from the build. Please use the `nextstrain build` command
+          locally to reattach to AWS Batch builds to download outputs.
```

Comment:
Ah, I was thinking we wouldn't want to be double billed for the job between GitHub and AWS. 
We probably will also easily hit the concurrent job limit if we leave long running jobs attached.

Reply:
We're not currently billed for our GitHub Actions usage because it falls under the gratis usage limits.

Leaving jobs attached _would_ impact concurrency, which I guess we do bump up against already, e.g. right now:

![image](https://github.com/nextstrain/.github/assets/79913/ddfaf2d3-42b2-4095-8d43-daf318db8872)

We could presumably self-host a runner, use it for the jobs which submit to AWS Batch (which consume very little actual compute), and then let those jobs stack up there without affecting our concurrency limits elsewhere.

Reply:
For example, for about $100/yr (committed to, but not paid upfront) we could self-host a single, always-on runner on a t4g.small EC2 instance with 2 CPUs and 2GB of RAM. We could also use other auto-scaling solutions and likely reduce the cost further while increasing capacity.

Reply:
I thought we get 2000min/month free and anything more would be billed?

Reply:
Yes… except we have a spending limit of $0/month applied to the account, which prevents that.

Reply:
Ah I see, so if we hit the free limit then all GitHub Actions would just stop running? I guess that's even more support to set up a self-hosted runner. Maybe we'd even want to move _all_ pathogen builds to the self-hosted runner if we plan to scale up more automated builds. 

I'm going to leave this as a future improvement though.

Reply:
Instead of self-hosting (at least for these reasons), we could also choose to upgrade our plan and get more concurrency and usage time if we want. I believe @victorlin was looking into this and determined we didn't really need it yet.

+1 for future improvement for sure.

Reply:
Oh, but one important point I forgot about is that the usage limits only apply to _private_ repos. Since nearly all of our repos are public, esp. ones using Actions, this means we're nowhere near our monthly allotment of 2k minutes.

Reply:
An [untested WIP](https://gist.github.com/tsibley/258ab1b1e76f0bb32dbc3c17f892c169) which I'll play with when I'm back from vacation.

Reply:
https://github.com/tsibley/blab-standup/blob/master/2023-08-18.md

Reply:
> Test in a Nextstrain-owned repo.  Which one?

If we don't want to pollute a pathogen repo's actions, we could point to a pathogen repo from within https://github.com/nextstrain/test-github-actions.

Reply:
Ahhh, I didn't recall that repo existing. I'm testing within https://github.com/nextstrain/.github for now.

Reply:
https://github.com/nextstrain/.github/actions/runs/5932617642/job/16086780919#step:5:1 :tada:

Reply:
Ha, if we're going to run these on our own runners though, we can avoid the awkward job chaining that works around GitHub's timeout and just set our timeouts longer. :upside_down_face:

Reply:
https://github.com/tsibley/blab-standup/blob/master/2023-08-22.md#22-august-2023